kitx.loadJSONSync(path.join(__dirname, '../../package.json')) 这一行导致的错误

错误堆栈
```
notificationsAlerts.ts:40 Error: ENOENT: no such file or directory, open 'c:\Users\xxx\yyy\zzzz\package.json'
    at Object.openSync (node:fs:585:3)
    at Object.func [as openSync] (node:electron/js2c/asar_bundle:5:1812)
    at Object.readFileSync (node:fs:453:35)
    at Object.readFileSync (node:electron/js2c/asar_bundle:5:9160)
    at Object.loadJSONSync (c:\Users\xxx\yyy\zzz\cloud-storage-dashboard\dist\webpack:\cloud-storage-dashboard\node_modules\kitx\lib\index.js:15:1)
at Object../node_modules/@alicloud/credentials/dist/util/helper.js (c:\Users\xxx\yyy\zzz\cloud-storage-dashboard\dist\webpack:\cloud-storage-dashboard\node_modules\@alicloud\credentials\dist\util\helper.js:29:1)
```
错误原因，代码被 webpack 编译出来之后由于编译出来的文件目录结构发生了变化，导致下面这一行找不到指定文件
```js
const pkg = kitx.loadJSONSync(path.join(__dirname, '../../package.json'));
```
建议改成直接 `import pkgJSON from '../../package.json'`

> 错误堆栈
> 
> ```
> notificationsAlerts.ts:40 Error: ENOENT: no such file or directory, open 'c:\Users\xxx\yyy\zzzz\package.json'
>     at Object.openSync (node:fs:585:3)
>     at Object.func [as openSync] (node:electron/js2c/asar_bundle:5:1812)
>     at Object.readFileSync (node:fs:453:35)
>     at Object.readFileSync (node:electron/js2c/asar_bundle:5:9160)
>     at Object.loadJSONSync (c:\Users\xxx\yyy\zzz\cloud-storage-dashboard\dist\webpack:\cloud-storage-dashboard\node_modules\kitx\lib\index.js:15:1)
> at Object../node_modules/@alicloud/credentials/dist/util/helper.js (c:\Users\xxx\yyy\zzz\cloud-storage-dashboard\dist\webpack:\cloud-storage-dashboard\node_modules\@alicloud\credentials\dist\util\helper.js:29:1)
> ```
> 
> 错误原因，代码被 webpack 编译出来之后由于编译出来的文件目录结构发生了变化，导致下面这一行找不到指定文件
> 
> ```js
> const pkg = kitx.loadJSONSync(path.join(__dirname, '../../package.json'));
> ```
> 
> 建议改成直接 `import pkgJSON from '../../package.json'`

你好，我这边也是报相似的错，请问我应该怎么处理？

用这个库 ![patch-package](https://www.npmjs.com/package/patch-package)，我把源代码按照上面的方式修改之后编译出来，然后复制到 node_modules 中，用 patch-package 把 node_modules 的更改记录下来

> > 错误堆栈
> > ```
> > notificationsAlerts.ts:40 Error: ENOENT: no such file or directory, open 'c:\Users\xxx\yyy\zzzz\package.json'
> >     at Object.openSync (node:fs:585:3)
> >     at Object.func [as openSync] (node:electron/js2c/asar_bundle:5:1812)
> >     at Object.readFileSync (node:fs:453:35)
> >     at Object.readFileSync (node:electron/js2c/asar_bundle:5:9160)
> >     at Object.loadJSONSync (c:\Users\xxx\yyy\zzz\cloud-storage-dashboard\dist\webpack:\cloud-storage-dashboard\node_modules\kitx\lib\index.js:15:1)
> > at Object../node_modules/@alicloud/credentials/dist/util/helper.js (c:\Users\xxx\yyy\zzz\cloud-storage-dashboard\dist\webpack:\cloud-storage-dashboard\node_modules\@alicloud\credentials\dist\util\helper.js:29:1)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 错误原因，代码被 webpack 编译出来之后由于编译出来的文件目录结构发生了变化，导致下面这一行找不到指定文件
> > ```js
> > const pkg = kitx.loadJSONSync(path.join(__dirname, '../../package.json'));
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 建议改成直接 `import pkgJSON from '../../package.json'`
> 
> 你好，我这边也是报相似的错，请问我应该怎么处理？

用这个库 ![patch-package](https://www.npmjs.com/package/patch-package)，我把源代码按照上面的方式修改之后编译出来，然后复制到 node_modules 中，用 patch-package 把 node_modules 的更改记录下来

> > > 错误堆栈
> > > ```
> > > notificationsAlerts.ts:40 Error: ENOENT: no such file or directory, open 'c:\Users\xxx\yyy\zzzz\package.json'
> > >     at Object.openSync (node:fs:585:3)
> > >     at Object.func [as openSync] (node:electron/js2c/asar_bundle:5:1812)
> > >     at Object.readFileSync (node:fs:453:35)
> > >     at Object.readFileSync (node:electron/js2c/asar_bundle:5:9160)
> > >     at Object.loadJSONSync (c:\Users\xxx\yyy\zzz\cloud-storage-dashboard\dist\webpack:\cloud-storage-dashboard\node_modules\kitx\lib\index.js:15:1)
> > > at Object../node_modules/@alicloud/credentials/dist/util/helper.js (c:\Users\xxx\yyy\zzz\cloud-storage-dashboard\dist\webpack:\cloud-storage-dashboard\node_modules\@alicloud\credentials\dist\util\helper.js:29:1)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 错误原因，代码被 webpack 编译出来之后由于编译出来的文件目录结构发生了变化，导致下面这一行找不到指定文件
> > > ```js
> > > const pkg = kitx.loadJSONSync(path.join(__dirname, '../../package.json'));
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 建议改成直接 `import pkgJSON from '../../package.json'`
> > 
> > 
> > 你好，我这边也是报相似的错，请问我应该怎么处理？
> 
> 用这个库 [patch-package](https://www.npmjs.com/package/patch-package)，我把源代码按照上面的方式修改之后编译出来，然后复制到 node_modules 中，用 patch-package 把 node_modules 的更改记录下来

按照你的做法，我把源码拉下来之后，改下package.json的引用方式后，发现构建产物多了个src目录。不知道什么情况。看你fork的仓库中，也有dist/src这个目录

--- a/src/util/helper.ts
+++ b/src/util/helper.ts
@@ -1,8 +1,5 @@
 import * as os from 'os';
-import kitx from 'kitx';
-import path from 'path';
-
-const pkg = kitx.loadJSONSync(path.join(__dirname, '../../package.json'));
+import pkg from '../../package.json';
 
 export const DEFAULT_UA = `AlibabaCloud (${os.platform()}; ${os.arch()}) ` +
     `Node.js/${process.version} Core/${pkg.version}`;

--- a/test/util/helper.test.ts
+++ b/test/util/helper.test.ts
@@ -1,0 +1,8 @@
+import {DEFAULT_UA} from '../../src/util/helper';
+import pkg from '../../package.json';
+import expect from 'expect.js';
+
+
+it('should having version field', () => {
+    expect(DEFAULT_UA).have.contain(pkg.version);
+})

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,22 +1,23 @@
 {
-    "compilerOptions": {
-        "declaration": true,
-        "module": "commonjs",
-        "esModuleInterop": true,
-        "target": "ES2017",
-        "noImplicitAny": true,
-        "moduleResolution": "node",
-        "sourceMap": true,
-        "outDir": "./dist",
-        "baseUrl": ".",
-        "paths": {
-            "*": [
-                "node_modules/*",
-                "src/types/*"
-            ]
-        }
-    },
-    "include": [
-        "src/**/*"
-    ]
+  "compilerOptions": {
+    "declaration": true,
+    "module": "commonjs",
+    "esModuleInterop": true,
+    "target": "ES2017",
+    "resolveJsonModule": true,
+    "noImplicitAny": true,
+    "moduleResolution": "node",
+    "sourceMap": true,
+    "outDir": "./dist",
+    "baseUrl": ".",
+    "paths": {
+      "*": [
+        "node_modules/*",
+        "src/types/*"
+      ]
+    }
+  },
+  "include": [
+    "src/**/*"
+  ]
 }

